cmd/grype-db/cli/commands/cache_backup.go

```diff
@@ -121,7 +124,7 @@ func cacheBackup(cfg cacheBackupConfig) error {
 	return nil
 }
 
-func archiveProvider(root string, name string, writer *tar.Writer) error {
+func archiveProvider(cfg cacheBackupConfig, root string, name string, writer *tar.Writer) error {
```

Comment:
there is probably enough logic here to warrant adding a test around this function (or a decomposed section of it)

Reply:
I was not able to get the test completed in time for this, see if this is acceptable in this form and i can add a test later